### PR TITLE
[minor] [MASCORE-1896] Allow db2ucluster addons and auditlogs PV to be set

### PR DESCRIPTION
**Pull request by twhart**
Added and tested out the new configuration on noble3 in our SaaS fyre environment. This allows us to configure the storage class and size for the audit logs in our db2u configuration

image

### CODE REVIEW

File: image/cli/mascli/functions/help/install_help [hunk 1]
```diff
@@ -91,11 +91,12 @@ Advanced Db2u Universal Operator Configuration - Resource Requests (Optional):
...
 Advanced Db2u Universal Operator Configuration - Storage (Optional):
-      --db2u-backup-storage ${COLOR_YELLOW}DB2_BACKUP_STORAGE_SIZE${TEXT_RESET}  Customize Db2 storage capacity
-      --db2u-data-storage ${COLOR_YELLOW}DB2_DATA_STORAGE_SIZE${TEXT_RESET}      Customize Db2 storage capacity
-      --db2u-logs-storage ${COLOR_YELLOW}DB2_LOGS_STORAGE_SIZE${TEXT_RESET}      Customize Db2 storage capacity
-      --db2u-meta-storage ${COLOR_YELLOW}DB2_META_STORAGE_SIZE${TEXT_RESET}      Customize Db2 storage capacity
-      --db2u-temp-storage ${COLOR_YELLOW}DB2_TEMP_STORAGE_SIZE${TEXT_RESET}      Customize Db2 storage capacity
+      --db2u-backup-storage ${COLOR_YELLOW}DB2_BACKUP_STORAGE_SIZE${TEXT_RESET}           Customize Db2 storage capacity
+      --db2u-data-storage ${COLOR_YELLOW}DB2_DATA_STORAGE_SIZE${TEXT_RESET}               Customize Db2 storage capacity
+      --db2u-logs-storage ${COLOR_YELLOW}DB2_LOGS_STORAGE_SIZE${TEXT_RESET}               Customize Db2 storage capacity
+      --db2u-audit-logs-storage ${COLOR_YELLOW}DB2_AUDIT_LOGS_STORAGE_SIZE${TEXT_RESET}   Customize Db2 storage capacity
```
Comment (whitfiea) [inline image/cli/mascli/functions/help/install_help]: This file isn't used in gitops so we should remove this change

File: docs/commands/install.md [hunk 2]
```diff
@@ -105,11 +105,12 @@ Usage
...
 ### Advanced Db2u Universal Operator Configuration - Storage (Optional):
-- `--db2u-backup-storage DB2_BACKUP_STORAGE_SIZE`    Customize Db2 storage capacity
-- `--db2u-data-storage DB2_DATA_STORAGE_SIZE`        Customize Db2 storage capacity
-- `--db2u-logs-storage DB2_LOGS_STORAGE_SIZE`        Customize Db2 storage capacity
-- `--db2u-meta-storage DB2_META_STORAGE_SIZE`        Customize Db2 storage capacity
-- `--db2u-temp-storage DB2_TEMP_STORAGE_SIZE`        Customize Db2 storage capacity
+- `--db2u-backup-storage DB2_BACKUP_STORAGE_SIZE`           Customize Db2 storage capacity
+- `--db2u-data-storage DB2_DATA_STORAGE_SIZE`               Customize Db2 storage capacity
+- `--db2u-logs-storage DB2_LOGS_STORAGE_SIZE`               Customize Db2 storage capacity
+- `--db2u-audit-logs-storage DB2_AUDIT_LOGS_STORAGE_SIZE`   Customize Db2 storage capacity
```
Comment (whitfiea) [inline docs/commands/install.md]: This file isn't used in gitops so we should remove this change

File: image/cli/mascli/functions/install [hunk 3]
```diff
@@ -246,6 +246,9 @@ function install_noninteractive() {
       --db2u-logs-storage)
         export DB2_LOGS_STORAGE_SIZE=$1 && shift
         ;;
+      --db2u-audit-logs-storage)
+        export DB2_AUDIT_LOGS_STORAGE_SIZE=$1 && shift
```
Comment (whitfiea) [inline image/cli/mascli/functions/install]: This file isn't used in gitops so we should remove this change

File: image/cli/mascli/functions/internal/install_config_db2 [hunk 4]
```diff
@@ -183,6 +183,7 @@ function db2u_configuration() {
     echo_reset_dim " - Backup:           ${COLOR_MAGENTA}${DB2_BACKUP_STORAGE_SIZE:-Default}"
     echo_reset_dim " - Temp:             ${COLOR_MAGENTA}${DB2_TEMP_STORAGE_SIZE:-Default}"
     echo_reset_dim " - Transaction Logs: ${COLOR_MAGENTA}${DB2_LOGS_STORAGE_SIZE:-Default}"
+    echo_reset_dim " - Audit Logs:       ${COLOR_MAGENTA}${DB2_AUDIT_LOGS_STORAGE_SIZE:-Default}"
```
Comment (whitfiea) [inline image/cli/mascli/functions/internal/install_config_db2]: This file isn't used in gitops so we should remove this change

File: image/cli/mascli/functions/internal/install_config_sno [hunk 5]
```diff
@@ -16,5 +16,6 @@ function install_config_sno() {
     export DB2_BACKUP_STORAGE_ACCESSMODE=ReadWriteOnce
     export DB2_DATA_STORAGE_ACCESSMODE=ReadWriteOnce
     export DB2_LOGS_STORAGE_ACCESSMODE=ReadWriteOnce
+    export DB2_AUDIT_LOGS_STORAGE_ACCESSMODE=ReadWriteOnce
```
Comment (whitfiea) [inline image/cli/mascli/functions/internal/install_config_sno]: This file isn't used in gitops so we should remove this change

File: image/cli/mascli/functions/internal/install_config_storage_classes [hunk 6]
```diff
@@ -18,6 +18,7 @@
 # - grafana_instance_storage_class [ibmc-block-gold, ocs-storagecluster-ceph-rbd, managed-premium]
 # - db2_data_storage_class [ibmc-block-gold, ocs-storagecluster-ceph-rbd, managed-premium]
 # - db2_logs_storage_class [ibmc-block-gold, ocs-storagecluster-ceph-rbd, managed-premium]
+# - db2_audit_logs_storage_class [ibmc-block-gold, ocs-storagecluster-ceph-rbd, managed-premium]
```
Comment (whitfiea) [inline image/cli/mascli/functions/internal/install_config_storage_classes]: This file isn't used in gitops so we should remove this change

File: image/cli/mascli/functions/internal/install_config_storage_classes_sno [hunk 7]
```diff
@@ -9,6 +9,7 @@
 # - grafana_instance_storage_class [ibmc-block-gold, ocs-storagecluster-ceph-rbd, managed-premium]
 # - db2_data_storage_class [ibmc-block-gold, ocs-storagecluster-ceph-rbd, managed-premium]
 # - db2_logs_storage_class [ibmc-block-gold, ocs-storagecluster-ceph-rbd, managed-premium]
+# - db2_audit_logs_storage_class [ibmc-block-gold, ocs-storagecluster-ceph-rbd, managed-premium]
```
Comment (whitfiea) [inline image/cli/mascli/functions/internal/install_config_storage_classes_sno]: This file isn't used in gitops so we should remove this change

File: image/cli/mascli/functions/internal/install_pipelinerun_preview [hunk 8]
```diff
@@ -201,6 +201,7 @@ function install_pipelinerun_preview() {
     echo_reset_dim "Backup Storage ............ ${COLOR_MAGENTA}${DB2_BACKUP_STORAGE_SIZE:-<default>}"
     echo_reset_dim "Temp Storage .............. ${COLOR_MAGENTA}${DB2_TEMP_STORAGE_SIZE:-<default>}"
     echo_reset_dim "Transaction Logs Storage .. ${COLOR_MAGENTA}${DB2_LOGS_STORAGE_SIZE:-<default>}"
+    echo_reset_dim "Transaction Audit Logs Storage .. ${COLOR_MAGENTA}${DB2_AUDIT_LOGS_STORAGE_SIZE:-<default>}"
```
Comment (whitfiea) [inline image/cli/mascli/functions/internal/install_pipelinerun_preview]: This file isn't used in gitops so we should remove this change

File: image/cli/mascli/functions/internal/save_config [hunk 9]
```diff
@@ -186,6 +186,7 @@ export DB2_META_STORAGE_SIZE=$DB2_META_STORAGE_SIZE
 export DB2_DATA_STORAGE_SIZE=$DB2_DATA_STORAGE_SIZE
 export DB2_BACKUP_STORAGE_SIZE=$DB2_BACKUP_STORAGE_SIZE
 export DB2_LOGS_STORAGE_SIZE=$DB2_LOGS_STORAGE_SIZE
+export DB2_AUDIT_LOGS_STORAGE_SIZE=$DB2_AUDIT_LOGS_STORAGE_SIZE
```
Comment (whitfiea) [inline image/cli/mascli/functions/internal/save_config]: This file isn't used in gitops so we should remove this change

File: image/cli/mascli/templates/pipelinerun-install.yaml [hunk 10]
```diff
@@ -88,6 +88,8 @@ spec:
       value: '$DB2_BACKUP_STORAGE_ACCESSMODE'
     - name: db2_logs_storage_accessmode
       value: '$DB2_LOGS_STORAGE_ACCESSMODE'
+    - name: db2_audit_logs_storage_accessmode
+      value: '$DB2_AUDIT_LOGS_STORAGE_ACCESSMODE'
```
Comment (whitfiea) [inline image/cli/mascli/templates/pipelinerun-install.yaml]: This file isn't used in gitops so we should remove this change

File: tekton/src/params/install-db2.yml.j2 [hunk 11]
```diff
@@ -108,6 +108,14 @@
   type: string
   description: Storage capacity (logs)
   default: "100Gi"
+- name: db2_audit_logs_storage_class
+  type: string
+  description: Storage class (audit logs)
+  default: ""
+- name: db2_audit_logs_storage_size
+  type: string
+  description: Storage capacity (audit logs)
```
Comment (whitfiea) [inline tekton/src/params/install-db2.yml.j2]: This file isn't used in gitops so we should remove this change

File: tekton/src/pipelines/taskdefs/dependencies/db2.yml.j2 [hunk 12]
```diff
@@ -90,6 +92,8 @@
       value: $(params.storage_class_rwx)
     - name: db2_logs_storage_class
       value: $(params.storage_class_rwo)
+    - name: db2_audit_logs_storage_class
+      value: $(params.storage_class_rwo)
```
Comment (whitfiea) [inline tekton/src/pipelines/taskdefs/dependencies/db2.yml.j2]: This file isn't used in gitops so we should remove this change

File: tekton/src/tasks/dependencies/db2.yml.j2 [hunk 13]
```diff
@@ -98,6 +98,9 @@ spec:
     - name: db2_logs_storage_size
       type: string
       default: ""
+    - name: db2_audit_logs_storage_size
+      type: string
+      default: ""
```
Comment (whitfiea) [inline tekton/src/tasks/dependencies/db2.yml.j2]: This file isn't used in gitops so we should remove this change

File: image/cli/mascli/functions/gitops_db2u_database [hunk 14]
```diff
@@ -347,6 +360,12 @@ function gitops_db2u_database() {
   if [[ -z $DB2_LOGS_STORAGE_ACCESSMODE ]]; then
     export DB2_LOGS_STORAGE_ACCESSMODE=ReadWriteOnce
   fi  
+  if [[ -z $DB2_AUDIT_LOGS_STORAGE_SIZE ]]; then
+    export DB2_AUDIT_LOGS_STORAGE_SIZE=100Gi
+  fi 
+  if [[ -z $DB2_AUDIT_LOGS_STORAGE_ACCESSMODE ]]; then
+    export DB2_AUDIT_LOGS_STORAGE_ACCESSMODE=ReadWriteOnce
```
Comment (whitfiea) [inline image/cli/mascli/functions/gitops_db2u_database]: According to https://ibmdocs-test.dcs.ibm.com/docs/en/SSQNUZ_4.6_test?topic=security-configuring-audit-logging this should be ReadWriteMany

**Review by whitfiea — CHANGES_REQUESTED** [review]

**Comment by twhart** [comment]
Tested out the audit config addon changes 

ReadWriteMany Change:
image

Build Configurations:
image

image


Build 
https://cloud.ibm.com/devops/pipelines/tekton/8bb11078-2a03-43b2-9d30-df3ea17af143/runs/df8ef5f0-2a59-4d1e-8eb7-30d04c541c48/gitops-db2u-database-manage/gitops-db2u-database?env_id=ibm:yp:us-south

File: docs/commands/install.md [hunk 15]
```diff
@@ -105,11 +105,11 @@ Usage
...
-- `--db2u-backup-storage DB2_BACKUP_STORAGE_SIZE`    Customize Db2 storage capacity
-- `--db2u-data-storage DB2_DATA_STORAGE_SIZE`        Customize Db2 storage capacity
-- `--db2u-logs-storage DB2_LOGS_STORAGE_SIZE`        Customize Db2 storage capacity
-- `--db2u-meta-storage DB2_META_STORAGE_SIZE`        Customize Db2 storage capacity
-- `--db2u-temp-storage DB2_TEMP_STORAGE_SIZE`        Customize Db2 storage capacity
+- `--db2u-backup-storage DB2_BACKUP_STORAGE_SIZE`           Customize Db2 storage capacity
+- `--db2u-data-storage DB2_DATA_STORAGE_SIZE`               Customize Db2 storage capacity
+- `--db2u-logs-storage DB2_LOGS_STORAGE_SIZE`               Customize Db2 storage capacity
+- `--db2u-meta-storage DB2_META_STORAGE_SIZE`               Customize Db2 storage capacity
+- `--db2u-temp-storage DB2_TEMP_STORAGE_SIZE`               Customize Db2 storage capacity
```
Comment (whitfiea) [inline docs/commands/install.md]: This file change should be removed as well.

File: image/cli/mascli/functions/gitops_db2u_database [hunk 16]
```diff
@@ -21,35 +21,39 @@ IBM Maximo Application Suite:
...
+      --db2-cpu-requests ${COLOR_YELLOW}DB2_CPU_REQUESTS${TEXT_RESET}                                         Define the Kubernetes CPU request for the Db2 pod.
+      --db2-cpu-limits ${COLOR_YELLOW}DB2_CPU_LIMITS${TEXT_RESET}                                             Define the Kubernetes CPU limit for the Db2 pod.
+      --db2-memory-requests ${COLOR_YELLOW}DB2_MEMORY_REQUESTS${TEXT_RESET}                                   Define the Kubernetes memory request for the Db2 pod.
+      --db2-memory-limits ${COLOR_YELLOW}DB2_MEMORY_LIMITS${TEXT_RESET}                                       Define the Kubernetes memory limit for the Db2 pod.
+      --db2-instance-registry-yaml ${COLOR_YELLOW}DB2_INSTANCE_REGISTRY_YAML${TEXT_RESET}                     Yaml file containing the db2ucluster instance registry section.
+      --db2-instance-dbm-config-yaml ${COLOR_YELLOW}DB2_INSTANCE_DBM_CONFIG_YAML${TEXT_RESET}                 Yaml file containing the db2ucluster instance dbm config section.
+      --db2-database-db-config-yaml ${COLOR_YELLOW}DB2_DATABASE_DB_CONFIG_YAML${TEXT_RESET}
+      --db2-addons-audit-config-yaml ${COLOR_YELLOW}DB2_ADDONS_AUDIT_CONFIG_YAML${TEXT_RESET}
```
Comment (whitfiea) [inline image/cli/mascli/functions/gitops_db2u_database]: Looks like we lost the descrioption for these two

File: image/cli/mascli/functions/gitops_db2u_database [hunk 17]
```diff
@@ -268,6 +284,7 @@ function gitops_db2u_database_noninteractive() {
   [[ -z "$DB2_DATA_STORAGE_CLASS" ]] && gitops_db2u_help "DB2_DATA_STORAGE_CLASS is not set"
   [[ -z "$DB2_BACKUP_STORAGE_CLASS" ]] && gitops_db2u_help "DB2_BACKUP_STORAGE_CLASS is not set"
   [[ -z "$DB2_LOGS_STORAGE_CLASS" ]] && gitops_db2u_help "DB2_LOGS_STORAGE_CLASS is not set"
+  [[ -z "$DB2_AUDIT_LOGS_STORAGE_CLASS" ]] && gitops_db2u_help "DB2_AUDIT_LOGS_STORAGE_CLASS is not set"
```
Comment (whitfiea) [inline image/cli/mascli/functions/gitops_db2u_database]: The audit logs are optional so we shouldn't have DB2_AUDIT_LOGS_STORAGE_CLASS as a requirement

File: image/cli/mascli/functions/help/install_help [hunk 18]
```diff
@@ -91,11 +91,11 @@ Advanced Db2u Universal Operator Configuration - Resource Requests (Optional):
...
-      --db2u-backup-storage ${COLOR_YELLOW}DB2_BACKUP_STORAGE_SIZE${TEXT_RESET}  Customize Db2 storage capacity
-      --db2u-data-storage ${COLOR_YELLOW}DB2_DATA_STORAGE_SIZE${TEXT_RESET}      Customize Db2 storage capacity
-      --db2u-logs-storage ${COLOR_YELLOW}DB2_LOGS_STORAGE_SIZE${TEXT_RESET}      Customize Db2 storage capacity
-      --db2u-meta-storage ${COLOR_YELLOW}DB2_META_STORAGE_SIZE${TEXT_RESET}      Customize Db2 storage capacity
-      --db2u-temp-storage ${COLOR_YELLOW}DB2_TEMP_STORAGE_SIZE${TEXT_RESET}      Customize Db2 storage capacity
+      --db2u-backup-storage ${COLOR_YELLOW}DB2_BACKUP_STORAGE_SIZE${TEXT_RESET}           Customize Db2 storage capacity
+      --db2u-data-storage ${COLOR_YELLOW}DB2_DATA_STORAGE_SIZE${TEXT_RESET}               Customize Db2 storage capacity
+      --db2u-logs-storage ${COLOR_YELLOW}DB2_LOGS_STORAGE_SIZE${TEXT_RESET}               Customize Db2 storage capacity
+      --db2u-meta-storage ${COLOR_YELLOW}DB2_META_STORAGE_SIZE${TEXT_RESET}               Customize Db2 storage capacity
+      --db2u-temp-storage ${COLOR_YELLOW}DB2_TEMP_STORAGE_SIZE${TEXT_RESET}               Customize Db2 storage capacity
```
Comment (whitfiea) [inline image/cli/mascli/functions/help/install_help]: We should remove this file change as it is not part of gitops

**Review by whitfiea — CHANGES_REQUESTED** [review]

**Review by whitfiea — CHANGES_REQUESTED** [review]
Can you remove the ibm-mas_devops.tar.gz from this PR as I don't believe it is needed

**Comment by whitfiea** [comment]
I updated the devops tar.gz to be the gitops one

**Review by whitfiea — APPROVED** [review]

**Pull request merged**